### PR TITLE
[WIP] Add generic PHY for the new SPI XIP core

### PR DESCRIPTION
**Pull request by piotr-binkowski**
This adds initial version of a generic PHY for the new SPI XIP core. It was tested on the NeTV2 board with single/dual/quad reads (address only on mosi for now).

Currently missing or untested features are:
- [ ] support for configuring core parameters based on selected chip
- [ ] support for sending address on 2/4 data pins (untested)
- [ ] support for octal mode chips
- [ ] 32-bit addressing (untested)
- [ ] DDR transfers
- [ ] option for software based access to the chip (configure registers, erase/program)

This depends on #440

### CODE REVIEW

File: litex/boards/platforms/netv2.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
         Subsignal("cs_n", Pins("T19")),
         Subsignal("mosi", Pins("P22")),
         Subsignal("miso", Pins("R22")),
-        Subsignal("vpp", Pins("P21")),
+        Subsignal("wp",   Pins("P21")),
```
Comment (mithro) [inline litex/boards/platforms/netv2.py]: Can you submit this change to other locations that have netv2.py boards?

File: litex/soc/cores/spi_xip/phy/generic.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+soft_oe_mask = 0b0001
+addr_oe_mask = {
+    1: 0b0001,
+    2: 0b0011,
+    4: 0b1111,
+}
+
+def GetConfig(flash=None):
+    if flash is None:
+        return (24, 8, 1, 1, 1, 0x0b)
```
Comment (mithro) [inline litex/soc/cores/spi_xip/phy/generic.py]: Magic numbers - what do these mean? Should it be returning a named tuple or some other similar structure?

File: test/test_spi_xip.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            dut.addr_ok = 0
+            dut.cmd_ok = 0
+
+            yield dut.sink.valid.eq(0)
+            yield dut.source.ready.eq(1)
+
+            while (yield dut.source.valid) == 0:
+                yield
+
+            if (yield dut.source.addr) == (0xcafe<<2): # address cmd
```
Comment (mithro) [inline test/test_spi_xip.py]: Make address command a named value.

**Comment by piotr-binkowski** [comment]
I've added some more comments/docstrings to the version in litex-hub/litespi#8 so I think that we can close this PR and continue work on this there

**Pull request closed without merging**